### PR TITLE
Role based pings

### PR DESCRIPTION
**Pull request by jabbors**
This PR supports role based ping as described in https://github.com/gnyman/flowdock-notifybot/issues/3 with the exception of instant pings with `@`.

### CODE REVIEW

**Comment by gnyman** [comment]
As you have the API key, instead of applying a big change to the current notifybot and risk breaking that functionality maybe we can put this up for testing somewhere else but disable the standard slow-pings to avoid duplicates (or use a different prefix for testing).

That would allow some "integration testing" before we deploy it.

File: main.go [hunk 1]
```diff
@@ -21,7 +21,8 @@ type Username string
 type config struct {
 	FlowdockAPIKey string `yaml:"flowdock_api_key"`
 	StoragePath    string `yaml:"storage_path"`
```
Comment (gnyman) [inline main.go:23]: Probably a good idea to make this configurable also

**Comment by jabbors** [comment]
Glad you would say that. I actually implement support for running it in debug mode, no pings sent just logging and tags. I even tested it a bit in the test flow.

**Comment by gnyman** [comment]
I did some testing of my own, as described above. Very nice, good work, notifybot is growing into a real thing :-)

2 things that pop into my mind when using it.

1. Tags vs messages is a bit unclear , some commands add tags some send a message
2. The messages/tag editing will probably create a lot of spam, this is advanced level but it might be worth looking into moving those to PM. Or maybe just telling people to do the changes in #test as the roles are global

**Review by gnyman — APPROVED** [review]

**Comment by gnyman** [comment]
It seems making it understand PM:s is as "easy" as enabling user=1 in the API

https://www.flowdock.com/api/streaming

I tried it, it almost worked out of the box. Current coding means it does not handle it nicely but at least we know it's possible. If you want to work on it feel free otherwise I'm happy to take a look also.

File: main.go (hunk 1 above)
Comment (jabbors) [reply main.go:23]: Yes, will make it even more persistant. I started on a new refactoring branch so I'll include it there.

**Comment by jabbors** [comment]
Closing old PRs for abandoned projects.

**Pull request closed without merging**